README.md

# h_alert_dialog

A new Flutter package project for simple and beautiful dialogs

## Usage

To use this package, add h_alert_dialog as a dependency in your pubspec.yaml file.
And add this import to your file.

```dart
import 'package:h_alert_dialog/h_alert_dialog.dart';
```

## ScreenShots
<p>
    <img src="https://github.com/hassan-thabet/H_alert_dialog/raw/main/screenshots/success.png" width="260" title="">
    <img src="https://github.com/hassan-thabet/H_alert_dialog/raw/main/screenshots/error.png" width="260" title="">
</p>
<p>
    <img src="https://github.com/hassan-thabet/H_alert_dialog/raw/main/screenshots/info.png" width="260" title="">
    <img src="https://github.com/hassan-thabet/H_alert_dialog/raw/main/screenshots/warning.png" width="260" title="">
</p>

## Example

<p>
    <img src="https://github.com/hassan-thabet/H_alert_dialog/raw/main/screenshots/example.jpg" width="900" title="">
</p>

## Code
```dart
HAlertDialog.showCustomAlertBox(
                      context: context,
                      timerInSeconds: 2,
                      backgroundColor: Colors.green,
                      title: 'Success',
                      description: 'The proccess done successfully',
                      icon: Icons.done,
                      iconSize: 32,
                      iconColor: Colors.green,
                      titleFontFamily: 'Raleway',
                      titleFontSize: 22,
                      titleFontColor: Colors.black54,
                      descriptionFontFamily: 'Raleway',
                      descriptionFontColor: Colors.black45,
                      descriptionFontSize: 18,
                    );
```




CHANGELOG.md

## [0.0.1] - 2022-11-3

* A new flutter custom alert dialog box package , It makes creating a dialog box easier and faster, with many features ,
* suitable for all types of projects

lib/h_alert_dialog.dart
library h_alert_dialog;

import 'package:flutter/material.dart';

class HAlertDialog {
  static Future showCustomAlertBox({
    required BuildContext context,
    required int timerInSeconds,
    required Color backgroundColor,
    required String title,
    required String description,
    required IconData icon,
    required int iconSize,
    required Color iconColor,
    required String titleFontFamily,
    required double titleFontSize,
    required Color titleFontColor,
    required Color descriptionFontColor,
    required String descriptionFontFamily,
    required double descriptionFontSize,
  }) {
    Future.delayed(Duration(seconds: timerInSeconds), () {
      Navigator.pop(context);
    });

    return showDialog(
        context: context,
        builder: (context) {
          return AlertDialog(
            contentPadding: const EdgeInsets.all(0),
            shape: const RoundedRectangleBorder(
              borderRadius: BorderRadius.all(Radius.circular(15)),
            ),
            content: Column(
              mainAxisSize: MainAxisSize.min,
              children: <Widget>[
                Container(
                  width: double.infinity,
                  decoration: BoxDecoration(
                    color: backgroundColor,
                    borderRadius: const BorderRadius.only(
                      topLeft: Radius.circular(15),
                      topRight: Radius.circular(15),
                    ),
                  ),
                  child: Padding(
                    padding: const EdgeInsets.symmetric(vertical: 24),
                    child: SizedBox(
                      height: 65,
                      width: 65,
                      child: CircleAvatar(
                        radius: 65,
                        backgroundColor: Colors.white,
                        child: Icon(
                          icon,
                          color: iconColor,
                          size: 32,
                        ),
                      ),
                    ),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(top: 22, bottom: 4),
                  child: Text(
                    title,
                    textAlign: TextAlign.center,
                    maxLines: 1,
                    style: TextStyle(
                        color: titleFontColor,
                        fontFamily: titleFontFamily,
                        fontSize: titleFontSize,
                        fontWeight: FontWeight.w700),
                  ),
                ),
                Padding(
                  padding: const EdgeInsets.only(bottom: 14),
                  child: Text(
                    description,
                    textAlign: TextAlign.center,
                    style: TextStyle(
                        color: descriptionFontColor,
                        fontFamily: descriptionFontFamily,
                        fontSize: descriptionFontSize,
                        fontWeight: FontWeight.w300),
                  ),
                ),
              ],
            ),
            elevation: 10,
          );
        });
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:h_alert_dialog/h_alert_dialog.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'flutter example screen',
      home: ExampleScreen(),
    );
  }
}

class ExampleScreen extends StatelessWidget {
  const ExampleScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            MaterialButton(
              onPressed: () {
                HAlertDialog.showCustomAlertBox(
                  context: context,
                  backgroundColor: Colors.green,
                  title: 'Success',
                  description:
                      'This is the message that will appear in case of success',
                  icon: Icons.done,
                  iconSize: 32,
                  iconColor: Colors.green,
                  titleFontFamily: 'Raleway',
                  titleFontSize: 22,
                  titleFontColor: Colors.black54,
                  descriptionFontFamily: 'Raleway',
                  descriptionFontColor: Colors.black45,
                  descriptionFontSize: 18,
                  timerInSeconds: 2,
                );
              },
              child: const Text('Show Success Message'),
            ),
            MaterialButton(
              onPressed: () {
                HAlertDialog.showCustomAlertBox(
                  context: context,
                  backgroundColor: Colors.red,
                  title: 'Error',
                  description:
                      'This is the message that will appear in case of Error',
                  icon: Icons.error_outline,
                  iconSize: 32,
                  iconColor: Colors.red,
                  titleFontFamily: 'Raleway',
                  titleFontSize: 22,
                  titleFontColor: Colors.black54,
                  descriptionFontFamily: 'Raleway',
                  descriptionFontColor: Colors.black45,
                  descriptionFontSize: 18,
                  timerInSeconds: 2,
                );
              },
              child: const Text('Show Error Message'),
            ),
            MaterialButton(
              onPressed: () {
                HAlertDialog.showCustomAlertBox(
                  context: context,
                  backgroundColor: Colors.blue,
                  title: 'Info',
                  description:
                      'This is the message that will appear in case of Info',
                  icon: Icons.info_outline,
                  iconSize: 32,
                  iconColor: Colors.blue,
                  titleFontFamily: 'Raleway',
                  titleFontSize: 22,
                  titleFontColor: Colors.black54,
                  descriptionFontFamily: 'Raleway',
                  descriptionFontColor: Colors.black45,
                  descriptionFontSize: 18,
                  timerInSeconds: 2,
                );
              },
              child: const Text('Show Info Message'),
            ),
            MaterialButton(
              onPressed: () {
                HAlertDialog.showCustomAlertBox(
                  context: context,
                  backgroundColor: Colors.yellow,
                  title: 'Warning',
                  description:
                      'This is the message that will appear in case of Warning',
                  icon: Icons.warning_amber_rounded,
                  iconSize: 32,
                  iconColor: Colors.yellow,
                  titleFontFamily: 'Raleway',
                  titleFontSize: 22,
                  titleFontColor: Colors.black54,
                  descriptionFontFamily: 'Raleway',
                  descriptionFontColor: Colors.black45,
                  descriptionFontSize: 18,
                  timerInSeconds: 2,
                );
              },
              child: const Text('Show Warning Message'),
            ),
          ],
        ),
      ),
    );
  }
}
